Websites/webkit.org/demos/webgpu/scripts/hello-triangle.js

```diff
@@ -28,24 +28,33 @@ async function helloTriangle() {
     const pipelineLayoutDesc = { bindGroupLayouts: [uniformBindGroupLayout] };
     const layout = device.createPipelineLayout(pipelineLayoutDesc);
 /*
-    FIXME: Use WGSL once compiler is brought up
+    FIXME: Add support for binary expressions to match the metal code
+*/
     const wgslSource = `
-                     @vertex fn vsmain(@builtin(vertex_index) VertexIndex: u32) -> @builtin(position) vec4<f32>
+                     struct Vertex {
+                         @builtin(position) Position: vec4<f32>,
+                         @location(0) color: vec4<f32>,
+                     }
+
+                     @vertex fn vsmain(@builtin(vertex_index) VertexIndex: u32) -> Vertex
                      {
                          var pos: array<vec2<f32>, 3> = array<vec2<f32>, 3>(
                              vec2<f32>( 0.0,  0.5),
                              vec2<f32>(-0.5, -0.5),
                              vec2<f32>( 0.5, -0.5)
                          );
-                         return vec4<f32>(pos[VertexIndex], 0.0, 1.0);
+                         var vout : Vertex;
+                         vout.Position = vec4<f32>(pos[VertexIndex], 0.0, 1.0);
+                         vout.color = vec4<f32>(pos[VertexIndex], 0.0, 1.0);
+                         return vout;
                      }
 
-                     @fragment fn fsmain() -> @location(0) vec4<f32>
+                     @fragment fn fsmain(in: Vertex) -> @location(0) vec4<f32>
                      {
-                         return vec4<f32>(1.0, 0.0, 0.0, 1.0);
+                         return in.color;
                      }
     `;
- */
+/*
```

Comment:
Ditto about leaving code commented; usually we either user it or delete it.

Reply:
All of these examples files have both the MSL and WGSL code, one which is commented out. Should I just delete the MSL code once we switched for the WGSL for each file?